src/tests/base.py

```diff
@@ -398,7 +398,37 @@ def assertTmpFileNotTouchedBy(self, filename: str, func: Callable,
                                  relpath(filename, self.tmpd.name))
 
 
-class CleanAWSLoginEnvironment(TempDir):
+class CleanEnvironment(unittest.TestCase):
+    """Mixin Class for clearing and setting environment variables."""
```

Comment:
Interesting code. I don't take as much advantage of mixin classes as I would like.

Reply:
Mixins come in handy. The test code is very classy and complex. I'm happy that I made the most of the class system from a learning perspective, but from a maintenance perspective I wish I had used [pytest](https://docs.pytest.org/en/7.1.x/) which is much more pythonic then the built in unittest framework which feels like java. [pytest](https://docs.pytest.org/en/7.1.x/) is based on functions and is just much easier to deal with even with dealing with complex tests where you want to keep everything nice and DRY.